cmd/vic-machine/validate/validator.go

```diff
@@ -119,7 +122,64 @@ func (v *Validator) GetResourcePool(input *data.Data) (*object.ResourcePool, err
 	return v.Session.Pool, nil
 }
 
-func (v *Validator) getConnection(input *data.Data) (bool, error) {
+func (v *Validator) GetVCH(input *data.Data) (*object.ResourcePool, error) {
+	rp, err := v.GetResourcePool(input)
+	if err != nil {
+		return nil, err
+	}
+	vchPath := fmt.Sprintf("%s/%s", rp.InventoryPath, input.DisplayName)
+	vchPool, err := v.Session.Finder.ResourcePool(v.Context, vchPath)
+	if err != nil {
+		log.Errorf("Failed to get VCH resource pool %s: %s", vchPath, err)
+		return nil, err
+	}
+	return vchPool, nil
+}
+
+func (v *Validator) GetVCHConfig(vch *object.ResourcePool, name string) (*metadata.VirtualContainerHostConfigSpec, error) {
+	var err error
+	var vms []*vm.VirtualMachine
+
+	rp := compute.NewResourcePool(v.Context, v.Session, vch.Reference())
+	if vms, err = rp.GetChildrenVMs(v.Context, v.Session); err != nil {
```

Comment:
I think we need more test for if that VM is a VCH or not. Instead of just check the VM name.
But with GetChildVM, this check can be done after the VM is found. Will change it.